### PR TITLE
add timeout for integration test

### PR DESCRIPTION
**Pull request by TingDaoK**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: integration-testing/http_client_test.py [hunk 1]
```diff
@@ -18,15 +18,21 @@
...
     # gather all stderr and stdout to a single string that we print only if things go wrong
     process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-    output = process.communicate()[0]
-
-    if process.returncode != 0:
+    try:
+        # set a 100 sec timeout
+        output = process.communicate(timeout=100)[0]
+    except subprocess.TimeoutExpired:
+        process.kill()
```
Comment (graebm) [inline integration-testing/http_client_test.py:25]: Hmm I was worried that we're not showing the logs in the case of timeout, but the docs say you can call .communicate() again, they actually recommend it:

[docs](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate)
> The child process is not killed if the timeout expires, so in order to cleanup properly a well-behaved application should kill the child process and finish communication

(Check out the code sample in the link)

Maybe we can still get the full output? something like:

```
bool timedout = False
try:
    output = process.communicate(timeout=TIMEOUT)[0]
except subprocess.TimedoutExpired:
    timedout = True
    process.kill()
    output = process.communicate()[0]

if process.returncode != 0:
    ... all the same printing as before...
    if timedout:
        print("Process killed due to timeout after {} secs".format(TIMEOUT))

Comment (TingDaoK) [reply integration-testing/http_client_test.py:28]: Yeah, I saw the doc. Hmm, agreed on we should finish the communication, but probably the return code is not informative as we killed the process manually. Still think we need another RuntimeError in the end.

Comment (graebm) [reply integration-testing/http_client_test.py:28]: I was assuming the return code would be non-zero if the process was killed. You'd have to test to find out. or just make it: `if timedout or process.returncode != 0:`

Comment (TingDaoK) [reply integration-testing/http_client_test.py:28]: yeah, I have already tested it using aws-crt-nodejs. It's `-9`, if I remember it correctly, but `if timedout or process.returncode != 0` will be a better choice

**Review by graebm — APPROVED** [review]

**Pull request merged**